test/checks/navigation/identical-links-same-purpose.js

```diff
@@ -0,0 +1,118 @@
+describe('identical-links-same-purpose tests', function() {
+	'use strict';
+
+	var fixture = document.getElementById('fixture');
+	var queryFixture = axe.testUtils.queryFixture;
+	var check = checks['identical-links-same-purpose'];
+	var checkContext = axe.testUtils.MockCheckContext();
+	var options = {};
+
+	afterEach(function() {
+		fixture.innerHTML = '';
+		checkContext.reset();
+	});
+
+	it('returns undefined when element does not have a resource (empty href)', function() {
+		var vNode = queryFixture('<a id="target" href="">Go to google.com</a>');
+		var actual = check.evaluate.call(
+			checkContext,
+			vNode.actualNode,
+			options,
+			vNode
+		);
+		assert.isUndefined(actual);
+		assert.isNull(checkContext._data);
+	});
+
+	it('returns undefined when element does not have a resource (onclick does not change location)', function() {
+		var vNode = queryFixture(
+			'<span id="target" role="link" tabindex="0" onclick="return false;">Link text</span>'
+		);
+		var actual = check.evaluate.call(
+			checkContext,
+			vNode.actualNode,
+			options,
+			vNode
+		);
+		assert.isUndefined(actual);
+		assert.isNull(checkContext._data);
+	});
+
+	it('returns true when element has location resource', function() {
+		var vNode = queryFixture(
+			'<a id="target" href="http://facebook.com">Follow us</a>'
+		);
+		var actual = check.evaluate.call(
+			checkContext,
+			vNode.actualNode,
+			options,
+			vNode
+		);
+		assert.isTrue(actual);
+		assert.hasAllKeys(checkContext._data, ['accessibleText', 'linkResource']);
+	});
+
+	it('returns true when element has location resource', function() {
+		var vNode = queryFixture(
+			'<span id="target" role="link" tabindex="0" onclick="location=\'/pages/index.html\'">Link text</span>'
+		);
+		var actual = check.evaluate.call(
+			checkContext,
+			vNode.actualNode,
+			options,
+			vNode
+		);
+		assert.isTrue(actual);
+		assert.hasAllKeys(checkContext._data, ['accessibleText', 'linkResource']);
+	});
+
+	describe('after', function() {
```

Comment:
For the sake of consistency, test the after method in a different file. All other things with `after` do so.